engine/resources/mount.go

```diff
@@ -0,0 +1,715 @@
+}
+
+// MountUID is a unique resource identifier.
+type MountUID struct {
+	engine.BaseUID
+	name string
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *MountUID) IFF(uid engine.ResUID) bool {
+	res, ok := uid.(*MountUID)
+	if !ok {
+		return false
+	}
+	return obj.name == res.name
+}
+
+// UIDs includes all params to make a unique identification of this object.
+// Most resources only return one although some resources can return multiple.
+func (obj *MountRes) UIDs() []engine.ResUID {
+	x := &MountUID{
+		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
+		name:    obj.Name(),
+	}
+	return []engine.ResUID{x}
+}
+
+// Cmp compares two resources and return if they are equivalent.
+func (obj *MountRes) Cmp(r engine.Res) error {
+	// we can only compare MountRes to others of the same resource kind
+	res, ok := r.(*MountRes)
+	if !ok {
+		return fmt.Errorf("not a %s", obj.Kind())
+	}
+	if obj.State != res.State {
+		return fmt.Errorf("the State differs")
+	}
+	if obj.Type != res.Type {
+		return fmt.Errorf("the Type differs")
+	}
+	if !strMapEq(obj.Options, res.Options) {
+		return fmt.Errorf("the Options differ")
+	}
+	if obj.Freq != res.Freq {
+		return fmt.Errorf("the Type differs")
+	}
+	if obj.PassNo != res.PassNo {
+		return fmt.Errorf("the PassNo differs")
+	}
+
+	return nil
+}
+
+// UnmarshalYAML is the custom unmarshal handler for this struct.
+// It is primarily useful for setting the defaults.
+func (obj *MountRes) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawRes MountRes // indirection to avoid infinite recursion
+
+	def := obj.Default()       // get the default
+	res, ok := def.(*MountRes) // put in the right format
+	if !ok {
+		return fmt.Errorf("could not convert to MountRes")
+	}
+	raw := rawRes(*res) // convert; the defaults go here
+
+	if err := unmarshal(&raw); err != nil {
+		return err
+	}
+
+	*obj = MountRes(raw) // restore from indirection with type conversion!
+	return nil
+}
+
+// defaultMntOps returns a map that sets the default mount options for fstab
+// mounts.
+func defaultMntOps() map[string]string {
+	return map[string]string{"defaults": ""}
+}
+
+// strMapEq returns true, if and only if the two provided maps are identical.
+func strMapEq(x, y map[string]string) bool {
+	if len(x) != len(y) {
+		return false
+	}
+	for k, v := range x {
+		if val, ok := x[k]; !ok || v != val {
+			return false
+		}
+	}
+	return true
+}
+
+// fstabEntryExists checks whether or not a given mount exists in the provided
+// fstab file.
+func fstabEntryExists(file string, mount *fstab.Mount) (bool, error) {
+	mounts, err := fstab.ParseFile(file)
+	if err != nil {
+		return false, err
+	}
+	for _, m := range mounts {
+		if m.Equals(mount) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// fstabEntryAdd adds the given mount to the provided fstab file.
+func fstabEntryAdd(file string, mount *fstab.Mount) error {
+	mounts, err := fstab.ParseFile(file)
```

Comment:
each use of fstab.ParseFile has the errors done differently. consistency please :)